src/Pinch/Internal/Value.hs

```diff
@@ -191,6 +194,7 @@ instance Hashable (Value a) where
       VInt64  x -> s `hashWithSalt` (6 :: Int) `hashWithSalt` x
       VList   x -> s `hashWithSalt` (7 :: Int) `hashWithSalt` x
       VMap    x -> s `hashWithSalt` (8 :: Int) `hashWithSalt` x
+      VNullMap  -> s `hashWithSalt` (8 :: Int)
```

Comment:
I'm not so certain about this; the problem is that if you distinguish these cases then you lose the ability to round-trip values. This would be extremely surprising to the user. Really, `VNullMap` is an implementation detail which should be hid as much as possible. Really the fact that it's user-visible at all is IMHO problematic.